src/api/configuration.cpp

```diff
@@ -384,6 +388,33 @@ namespace mamba
         }
     }
 
+    fs::path get_conda_root_prefix()
+    {
+        std::vector<std::string> args = { "conda", "config", "--show", "root_prefix", "--json" };
+        std::string out, err;
+        auto [status, ec] = reproc::run(
+            args, reproc::options{}, reproc::sink::string(out), reproc::sink::string(err));
+
+        if (ec)
+        {
+            LOG_ERROR << "Conda root prefix not found using 'conda config' command";
+            throw std::runtime_error("Aborting.");
+        }
+        else
+        {
+            auto j = nlohmann::json::parse(out);
+            return j.at("root_prefix").get<std::string>();
+        }
+    }
+
+    void use_conda_root_prefix(bool force)
+    {
+        if (!Configuration::instance().at("root_prefix").configured() || force)
+        {
+            env::set("MAMBA_ROOT_PREFIX", get_conda_root_prefix());
```

Comment:
did you properly test this? This is not re-configuring the config? Is it properly propagated to the context?